Add plugin to turn aftertouch into CC

jonkerj opened the pull request:
This plugin turns (monophonic) channel pressure status messages (better known as Aftertouch) into continous controller messages. I use it to turn the aftertouch of my MIDI keyboard into a mod wheel, since a lot of synths do not implement AT.

jonkerj committed:
Add plugin to turn aftertouch into CC

This plugin turns (monophonic) channel pressure status messages (better
known as Aftertouch) into continous controller messages.

Signed-off-by: Jorik Jonker <jorik@kippendief.biz>

SpotlightKid commented on plugins/MIDIPressureToCC/DistrhoPluginInfo.h:
```diff
@@ -0,0 +1,46 @@
+/*
+ * MIDI PBToCC plugin based on DISTRHO Plugin Framework (DPF)
```
"MIDI PBToCC" -> "MIDI PressureToCC"

SpotlightKid commented on plugins/MIDIPressureToCC/DistrhoPluginInfo.h:
```diff
@@ -0,0 +1,46 @@
+ *
+ * SPDX-License-Identifier: MIT
+ *
+ * Copyright (C) 2019 Christopher Arndt <info@chrisarndt.de>
```
Copyright (C) 2020 Christopher Arndt <info@chrisarndt.de>, Jorik Jonker

SpotlightKid commented on plugins/MIDIPressureToCC/DistrhoPluginInfo.h:
```diff
@@ -0,0 +1,46 @@
+#define DISTRHO_PLUGIN_INFO_H
+
+#define DISTRHO_PLUGIN_BRAND        "chrisarndt.de"
+#define DISTRHO_PLUGIN_NAME         "MIDI Channel Pressure (Aftertouch) to CC"
```
That's a bit too long (This is gonna show up in the Hosts's plugin list).

Recommendation: "MIDI Pressure to CC"

Unfortunately, DPF doesn't support plugin descriptions via `rdfs:comment` [yet](https://github.com/DISTRHO/DPF/issues/230), but it's there in the `getDescription` method, if and when it does.

For now, let's just put the additional keywords in the description in the README.

SpotlightKid commented on plugins/MIDIPressureToCC/PluginMIDIPressureToCC.cpp:
```diff
@@ -0,0 +1,214 @@
+/*
+ * MIDI PBToCC plugin based on DISTRHO Plugin Framework (DPF)
```
Same comments as for the `DistrhoPluginInfo.h` file header apply.

SpotlightKid commented on plugins/MIDIPressureToCC/PluginMIDIPressureToCC.cpp:
```diff
@@ -0,0 +1,214 @@
+            parameter.ranges.max = 1;
+            break;
+        case paramCC:
+            parameter.name = "Target CC";
```
Please call this "Destination CC", as I have another plugin coming up, where I use that term.

SpotlightKid commented on plugins/MIDIPressureToCC/PluginMIDIPressureToCC.cpp:
```diff
@@ -0,0 +1,214 @@
+            break;
+        case paramCC:
+            parameter.name = "Target CC";
+            parameter.symbol = "cc";
```
Rename to `dest_cc` for clarity.

SpotlightKid commented on plugins/MIDIPressureToCC/PluginMIDIPressureToCC.cpp:
```diff
@@ -0,0 +1,214 @@
+        if (filterChannel == -1 || chan == filterChannel) {
+            cc_event.frame = events[i].frame;
+            cc_event.size = 3;
+	    cc_event.data[0] = MIDI_CONTROL_CHANGE | chan;
```
No tabs please.

SpotlightKid commented on plugins/MIDIPressureToCC/PluginMIDIPressureToCC.cpp:
```diff
@@ -0,0 +1,214 @@
+
+
+void PluginMIDIPressureToCC::run(const float**, float**, uint32_t,
+                           const MidiEvent* events, uint32_t eventCount) {
```
Please line up indentation, i.e. `const`, below `const`.

SpotlightKid commented on plugins/MIDIPressureToCC/PluginMIDIPressureToCC.hpp:
```diff
@@ -0,0 +1,162 @@
+ *
+ * SPDX-License-Identifier: MIT
+ *
+ * Copyright (C) 2019 Christopher Arndt <info@chrisarndt.de>
```
See 2nd comment for the `DistrhoPluginInfo.h` file header.

SpotlightKid commented on plugins/MIDIPressureToCC/PluginMIDIPressureToCC.hpp:
```diff
@@ -0,0 +1,162 @@
+ */
+
+#ifndef PLUGIN_MIDIPressureTOCC_H
+#define PLUGIN_MIDIPressureTOCC_H
```
Please use ALL_CAPS for the inclusion guard.

SpotlightKid commented on plugins/MIDIPressureToCC/PluginMIDIPressureToCC.hpp:
```diff
@@ -0,0 +1,162 @@
+    enum Parameters {
+        paramFilterChannel,
+        paramKeepOriginal,
+        paramCC,
```
Please rename to `paramDestCC` for clarity.

Don't forget to fix other occurrences too.

SpotlightKid commented on plugins/MIDIPressureToCC/PluginMIDIPressureToCC.hpp:
```diff
@@ -0,0 +1,162 @@
+    }
+
+    const char* getDescription() const override {
+        return "Convert Monophonic Channel Pressure (Aftertouch) into Control Change messages";
```
I wouldn't use title case here, so "monophonic" should be lower case as it's not part of the proper event name "Channel Pressure".

SpotlightKid commented on plugins/MIDIPressureToCC/PluginMIDIPressureToCC.hpp:
```diff
@@ -0,0 +1,162 @@
+    }
+
+    uint32_t getVersion() const noexcept override {
+        return d_version(1, 2, 2);
```
Please start with version 1.0.0, I'll up the version to one indicating stable one, when I make a release.

SpotlightKid commented on plugins/MIDIPressureToCC/PluginMIDIPressureToCC.hpp:
```diff
@@ -0,0 +1,162 @@
+        "Breath",
+        {0.0, 0.0, 2.0}
+    },
+    {
```
Maybe add a preset for "Foot Controller" too?

SpotlightKid reviewed (changes requested):
Many thanks for the PR! Looks good.

Please add a section for the plugin to the README (behind MIDIPBToCC):

```
### MIDI PressureToCC

Convert (monophonic) Channel Presszre (Aftertouch) into Control Change messages

![MIDI PressureToCC screenshot](screenshots/MIDIPressureToCC.png)

... short description ..
```

I'll do the screenshot, so that the style matches the other ones.

jonkerj committed:
Cosmetic changes to match rest of the project

Project author provided detailed feedback in PR #2, all of which made
good sense.  This commit should fix it all.

Signed-off-by: Jorik Jonker <jorik@kippendief.biz>

jonkerj commented:
All comments made sense, most of them were due to hasty replacements on my side. I think I've fixed them all, but you should probably check :-)

SpotlightKid commented on README.md:
```diff
@@ -57,6 +57,21 @@ Convert Pitch Bend into Control Change messages
+* Configurable destination controller (0-127): modulation, breath, expression,
+  etc.
+* Any unconverted messages are kept in the plugin's output.
+* Original Pitch Bend messages in the input range can be optionally kept as
```
s/Pitch Bend/Channel Pressure/

SpotlightKid replied:
d/in the input range/

SpotlightKid commented:
@jonkerj Can you fix the readme? Then I'll merge rightaway.

jonkerj commented:
It is fixed

SpotlightKid commented:
Still says "Pitch Bend" instead of "Channel Pressure".

jonkerj commented:
I tried to do this while having a meeting, sorry for my sloppyness. Force-pushed, hopefully OK now. Thanks for being patient and careful :-)

SpotlightKid commented:
Thanks for your contribution!

I also added a screenshot and listed you in the authors section of the readme.
